proposals/devcontainer-collections.md

```diff
@@ -0,0 +1,110 @@
+
+This files would ideally be created automatically by a process that automatically adds information like: checksums, signatures, download links, etc.
+
+Clients would be able to select different sources for collections while also giving the option to users to add their own sources.
+
+## Collection file (devcontainer-collection.json)
+
+- Created as a release of a repository that either contains or represents groups of definitions/features from the same user/owner/publisher. (e.g. same organization in Github.)
+- Contains metadata for each definition/feature, repository origin, download link, checksum or signing information.
+- All referenced definitions/features would belong to the same user/organization. (e.g. download links would be the same organization in GitHub.)
+- References exact releases of each definition/feature.
+
+| Property | Type | Description |
+| :--- | :--- | :--- |
+| publisher | string | The organization/user that publishes this group of definitions/features. Must match with the repository user in github.|
+| repository | string | The repository where this collection is contained.|
+| version | string | Version information of this collection list.|
+| features | array | The list of features that are contained in this collection. Same information as the metadata file for the feature.|
+| definitions | array | The list of definitions contained in this collection. Same information as the metadata file for the definition.|
+
+Creating the collections file would make some changes over the metadata of the features/definitions where they are generated.
+
+- Ids would get the collection repository added to them. (e.g. for 'https://github.com/community/features' then 'community/features' to 'myfeature1' for a feature with that id)
+- The links to the release files would be added.
+- Date.
+
+The same way creating the definitions/feature file would also be part of the release process for it so that the following metadata can be added:
+- Version.
+- Checksum.
+- Signatures.
+- Date.
+
+## Definitions/features (devcontainer-feature.json/devcontainer-definition.json)
+
+### Common Properties
+
+| Property | Type | Description |
+| :--- | :--- | :--- |
+| id | string | Id of the feature/definition. The ids should be unique in the contest of the repository where they exist. |
+| displayName | string | Name of the feature/definition |
+| shortDescription | string | Optional one sentence description |
+| description | string | Description of the feature/definition |
+| repositoryPath | string | Path to the repository and folder that contains the root of the code for the feature/definition. |
+| keywords | array | List of keywords relevant to a user that would search for this definition/feature. |
+| version | string | Versioning information for the feature/definition. |
+| files | array | List of links for the download files with checksums and optional signature information. |
+
```

Comment:
I mentioned this in the issue, but I'd suggest there should be a `metadata` property here as well for free-form data. Today there's things in these definition files that are specific to MS policies that could land under this category, and its likely other organizations have the same needs, but in different forms. Either that or we probably need to introduce the idea of a "bill of material" to track things that are added. There's existing formats for this already.

Reply:
Added the `metadata` property.